bridge-history-api/internal/logic/l1_fetcher.go

```diff
@@ -45,11 +45,14 @@ type L1FetcherLogic struct {
 	crossMessageOrm *orm.CrossMessage
 	batchEventOrm   *orm.BatchEvent
 
+	lastFinalizedBatchIndex            *uint64
+	nextPendingLastFinalizedBatchIndex uint64
```

Comment:
An alternative solution to achieve minimum changes:
1. we can remove these two variables and just use `batch_index <= height && batch_status != finalized` to update the db. thus no need to reload them when restarting or maintaining the update logic.
2. not splitting FinalizeBatch into multiple events in the following code (changes in this PR):
```
			var startBatchIndex uint64 = 0
			if lastFinalizedBatchIndex, ok := f.getLastFinalizedBatchIndex(); ok {
				startBatchIndex = lastFinalizedBatchIndex + 1
			}
			for index := startBatchIndex; index <= event.BatchIndex.Uint64(); index++ {
				l1BatchEvents = append(l1BatchEvents, &orm.BatchEvent{
					BatchStatus: int(btypes.BatchStatusTypeFinalized),
					BatchIndex:  index,
				})
			}
			f.setNextPendingLastFinalizedBatchIndex(event.BatchIndex.Uint64())
```
cc @georgehao @Thegaram

Reply:
There is a bug in current version, I will fix it soon integrating colin's version

Reply:
I have turned to your approach, how about it this time

Reply:
the implementation looks good. I also have not considered the deduplication of <= updates previously.